### PR TITLE
Advertise switch to maintainer teams and credit maintainers.

### PR DESCRIPTION
**Pull request by Zimmi48**
**Kind:** documentation

After the Coqdev session at the Coq workshop on Monday, I realized that we had forgotten to mention one major change that occurred during the 8.12 release cycle, namely the switch to maintainer teams and the important expansion of the number of maintainers. Given the role that reviewers play, it would be a shame not to thank them explicitly!

### CODE REVIEW

File: doc/sphinx/changes.rst
Comment (jfehrle) [inline doc/sphinx/changes.rst:73]:
```suggestion
Prior to 8.12, the code maintainers for each component of |Coq| were defined
directly in the file `CODEOWNERS` (such as @Zimmi48), which required a pull request to update it.
In 8.12,  `#11295<https://github.com/coq/coq/pull/11295>`_ now defines code maintainers
with GitHub teams (such as @coq/doc-maintainers) that can be updated more easily and
are visible in GitHub tooltips.  This provided an opportunity to significantly expand the
list of maintainters.  Special thanks to them!  Our current 31
```

> to adopt the idea that
every component of the system should have a dedicated team of
maintainers (in charge of reviewing and merging incoming pull
requests).

This part isn't new, is it?

**Review by jfehrle — APPROVED** [review]
I recommend changing that paragraph, which I'll leave to you rather than marking this as "Request Changes".

File: doc/sphinx/changes.rst
Comment (Zimmi48) [reply doc/sphinx/changes.rst:73]: This change goes beyond just the easiness of adding people to the maintainer teams: before this change each component (except those that had already adopted teams) had a *single* principal maintainer and a *single* secondary maintainer. This meant that most often, if principal maintainers were not available or responsive, PRs could get ignored. The maintainer teams idea puts team members on equal footing and ensures that more people will be made aware of incoming PRs (and thus they are less likely to get ignored). It also introduces the culture of expanding the teams as soon as you have a competent contributor who is relevant to maintain a specific component, instead of looking for new maintainers only when a component is badly maintained because its current principal maintainer doesn't have enough time for this anymore.

I'm not against slightly tweaking the phrasing, but it should remain short and to the point, rather than entering into technical details.

Comment (jfehrle) [reply doc/sphinx/changes.rst:73]: I wasn't part of the conversation.  I thought doc-maintainers already existed in some form prior to the PR but maybe I'm mistaken.

Seems to me the key point is changing from a single designated maintainer to a team of maintainers.

"to adopt the idea that every component of the system should have a dedicated team of
maintainers (in charge of reviewing and merging incoming pull requests)" doesn't seem very new.

Comment (Zimmi48) [reply doc/sphinx/changes.rst:73]: doc-maintainers already existed in the current form prior to the PR. This switch was just the decision to adopt such teams for *every* component.

>Seems to me the key point is changing from a single designated maintainer to a team of maintainers.

Yes!

>"to adopt the idea that every component of the system should have a dedicated team of
maintainers (in charge of reviewing and merging incoming pull requests)" doesn't seem very new.

OK, I'll try to tweak the wording.

**Comment by Zimmi48** [comment]
@jfehrle I've pushed a new version. Hopefully, it should address your concerns.

File: doc/sphinx/changes.rst
Comment (jfehrle) [inline doc/sphinx/changes.rst:68]:
```suggestion
Previously, most components of Coq had a single principal maintainer.
This was changed in 8.12 (`#11295
```

File: doc/sphinx/changes.rst
Comment (jfehrle) [inline doc/sphinx/changes.rst:71]:
```suggestion
<https://github.com/coq/coq/pull/11295>`_) so that every component now has
a team of maintainers, who are in charge of reviewing and
merging incoming pull requests.  This gave us a chance to
```

**Review by jfehrle — APPROVED** [review]
Small tweaks.

**Comment by ejgallego** [comment]
@Zimmi48 want to update this before merge?

**Comment by Zimmi48** [comment]
Yes, thanks for reminding me. I just applied the review suggestions.

**Pull request merged**
